spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWriteBuilder.java

```diff
@@ -81,6 +82,11 @@ public DeltaWrite build() {
         handleTimestampWithoutZone || !SparkUtil.hasTimestampWithoutZone(table.schema()),
         SparkUtil.TIMESTAMP_WITHOUT_TIMEZONE_ERROR);
 
+    if (writeConf.branch().equalsIgnoreCase(SnapshotRef.MAIN_BRANCH)) {
+      throw new UnsupportedOperationException(
+          "Row level operations on non-main branches are currently unsupported");
+    }
```

Comment:
@rdblue @aokolnychyi @namrathamyske So we'll throw (temporarily) if the branch option is set to a non-main branch.

Reply:
Need to see a good place to add a test for this...